db/migrate/20160317194215_remove_miq_user_role_from_miq_groups.rb

```diff
@@ -1,5 +1,18 @@
 class RemoveMiqUserRoleFromMiqGroups < ActiveRecord::Migration[5.0]
-  def change
-    remove_column :miq_groups, :miq_user_role_id, :bigint
+  class MiqGroup < ActiveRecord::Base
+    has_one :entitlement
+  end
+
+  def up
+    remove_column :miq_groups, :miq_user_role_id
+  end
+
+  def down
+    add_column :miq_groups, :miq_user_role_id, :bigint
+
+    MiqGroup.includes(:entitlement).where.not(:entitlements => {:miq_user_role_id => nil}).find_each do |group|
```

Comment:
Maybe you can write a down migration spec and see what happens if you delete app/models/entitlement.rb locally :wink: